lib/checkout.ts

```diff
@@ -0,0 +1,171 @@
+
+export const trialDays = 7;
+
+export const plans = [
+    {
+        name: 'Standard',
+        price: '40.00',
+        description: `Start out with inventory tracking, product variant lists,
+            and more. Perfect for businesses with fewer than 1,000 SKUs.`,
+        popular: false,
+        pid: '1',
+    },
+    {
+        name: 'Plus',
+        price: '79.95',
+        description: `Take your business to the next level with alerts and smart tracking.
+            Best for stores with more than 1,000 SKUs.`,
+        popular: true,
+        pid: '2'
+    },
+];
+
+// TODO: place data inside of a DB and create an API call
+const planItems = {
+    '1': {
+        product: {
+            id: '1',
+            type: 'app',
+        },
+        pricingPlan: {
+            interval: "MONTH",
+            price: {
+                value: 40.00,
+                currencyCode: 'USD'
+            },
+        },
+        trial_days: trialDays,
+    },
+    '2': {
+        product: {
+            id: '2',
+            type: 'app',
+        },
+        pricingPlan: {
+            interval: 'MONTH',
+            price: {
+                value: 79.95,
+                currencyCode: 'USD',
+            },
+        },
+        trial_days: trialDays,
+    },
+};
+
+const checkoutGraphQuery = `
+  mutation ($checkout: CreateCheckoutInput!) {
+    checkout {
+      createCheckout(input: $checkout) {
+        checkout {
+          id
+          accountId
+          status
+          checkoutUrl
+          items(first: 2) {
+            edges {
+              node {
+                subscriptionId
+                status
+                product {
+                  entityId
+                  type
+                }
+                scope {
+                  entityId
+                  type
+                }
+                pricingPlan {
+                  interval
+                  price {
+                    value
+                    currencyCode
+                  }
+                  trialDays
+                }
+                redirectUrl
+                description
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export function getCheckoutBody(productId: string, storeHash: string) {
+    return {
+        query: checkoutGraphQuery,
+        variables: {
+            checkout: {
+                accountId: `bc/account/account/${merchantUuid}`,
+                items: [
+                    {
+                    product: {
+                        entityId: `${checkoutAppId}`,
+                        type: 'APPLICATION'
+                    },
+                    scope: {
+                        entityId: `${storeHash}`,
+                        type: 'STORE'
+                    },
+                    pricingPlan: planItems[productId].pricingPlan,
+                    redirectUrl: `https://store-${storeHash}.my-staging.zone/manage/app/34030/${appId}/${productId}`,
```

Comment:
Should we make this an env variable so it can work in arbitrary environments?

Reply:
ahhh might be tricky if only because `.env` doesn't know the storeHash (unless we manually add it). Conversely, we can add a new variable that could capture the environment (e.g. staging)... I might go this route as we can then reuse the environment variable for api and login vars